src/generators.rs

```diff
@@ -907,156 +868,193 @@ pub fn directed_heavy_square_graph(
+///   from rustworkx.visualization import mpl_draw
 ///
-///   graph = rustworkx.generators.heavy_hex_graph(3)
-///   graphviz_draw(graph, lambda node: dict(
-///           color='black', fillcolor='lightblue', style='filled'))
+///   graph = rustworkx.generators.directed_binomial_tree_graph(4)
+///   mpl_draw(graph)
 ///
-#[pyfunction(multigraph = true)]
-#[pyo3(text_signature = "(d, /, multigraph=True)")]
-pub fn heavy_hex_graph(py: Python, d: usize, multigraph: bool) -> PyResult<graph::PyGraph> {
+#[pyfunction(bidirectional = "false", multigraph = "true")]
+#[pyo3(text_signature = "(order, /,  weights=None, bidirectional=False, multigraph=True)")]
+pub fn directed_binomial_tree_graph(
+    py: Python,
+    order: u32,
+    weights: Option<Vec<PyObject>>,
+    bidirectional: bool,
+    multigraph: bool,
+) -> PyResult<digraph::PyDiGraph> {
+    if order >= MAX_ORDER {
+        return Err(PyOverflowError::new_err(format!(
+            "An order of {} exceeds the max allowable size",
+            order
+        )));
+    }
     let default_fn = || py.None();
-    let graph: StablePyGraph<Undirected> =
-        match core_generators::heavy_hex_graph(d, default_fn, default_fn, false) {
-            Ok(graph) => graph,
-            Err(_) => return Err(PyIndexError::new_err("d must be an odd number.")),
-        };
-    Ok(graph::PyGraph {
+    let graph: StablePyGraph<Directed> = match core_generators::binomial_tree_graph(
+        order,
+        weights,
+        default_fn,
+        default_fn,
+        bidirectional,
+    ) {
+        Ok(graph) => graph,
+        Err(_) => {
+            return Err(PyIndexError::new_err(
+                "order and weights list not specified",
+            ))
+        }
+    };
+    Ok(digraph::PyDiGraph {
         graph,
         node_removed: false,
+        check_cycle: false,
+        cycle_state: algo::DfsSpace::default(),
         multigraph,
         attrs: py.None(),
     })
 }
 
-/// Generate a directed heavy hex graph. Fig. 2 of
-/// https://arxiv.org/abs/1907.09528
-/// An ASCII diagram of the graph is given by:
-///
-/// .. code-block:: text
-///
-///     ... D-S-D   D ...
-///         |   |   |
-///     ...-F   F-S-F ...
-///         |   |   |
-///     ... D   D   D ...
-///         |   |   |
-///     ... F-S-F   F-...
-///         |   |   |
-///         .........
-///         |   |   |
-///     ... D   D   D ...
-///         |   |   |
-///     ...-F   F-S-F ...
-///         |   |   |
-///     ... D   D   D ...
-///         |   |   |
-///     ... F-S-F   F-...
-///         |   |   |
-///         .........
-///         |   |   |
-///     ... D   D   D ...
-///         |   |   |
-///     ...-F   F-S-F ...
-///         |   |   |
-///     ... D   D   D ...
-///         |   |   |
-///     ... F-S-F   F-...
-///         |   |   |
-///     ... D   D-S-D ...
-///
+/// Creates a full r-ary tree of `n` nodes.
+/// Sometimes called a k-ary, n-ary, or m-ary tree.
```

Comment:
This might need a blank line, I think sphinx might treat these 2 lines as a single summary without it.